perceval/backends/remo.py

```diff
@@ -0,0 +1,275 @@
+                events = events_data['objects']
+            except:
+                cause = ("Bad JSON format for mozilla_events: %s" % (events_data))
+                raise ParseError(cause=cause)
+
+            for rep in events:
+                yield rep
+                nevents += 1
+
+            self._flush_cache_queue()
+
+        logger.info("Total number of events: %i (%i expected)", nevents, tevents)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the events from the cache.
+
+        :returns: a generator of events
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached bugs: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        nevents = 0
+
+        for items in cache_items:
+            events = json.loads(items)['objects']
+            for event in events:
+                nevents += 1
+                yield event
+
+        logger.info("Retrieval process completed: %s bugs retrieved from cache",
+                    nevents)
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+        return str(item['event_url'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a ReMo item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return float(parser.parse(item['end']).timestamp())
+
+
+class ReMoClient:
+    """ReMo API client.
+
+    This class implements a simple client to retrieve events from
+    projects in a ReMo site.
+
+    :param url: URL of ReMo (sample https://reps.mozilla.org)
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    MAX_ITEMS_EV = "40"  # max number of events per iteration
+
+    def __init__(self, url):
+        self.url = url
+        self.api_events_url = urljoin(self.url, '/api/v1/event/')
+        self.api_events_url += '/'  # API needs a final /
+
+    def call(self, params):
+        """Run an API command.
+        :param cgi: cgi command to run on the server
+        :param params: dict with the HTTP parameters needed to run
+            the given command
+        """
+        logger.debug("ReMo client calls API: %s params: %s",
+                     self.api_events_url, str(params))
+
+        req = requests.get(self.api_events_url, params=params)
+        req.raise_for_status()
+
+        return req.text
+
+    def get_events(self,  next_url=None):
+        """Retrieve all events"""
+        offset = "0"
+        if next_url:
+            # /api/v1/event/?limit=40&offset=40
+            offset = next_url.split("offset=")[1]
```

Comment:
Because this is specific logic for this API REST, so I find it more natural to have it in the client, and don't put the parsing logic in the user of the client.